frontend/endgame.py

```diff
@@ -0,0 +1,217 @@
+        }
+        self.profiles[players[2][0]] = {
+            "profile": Image(self.screen, self.profile_image, self.x_percent(1026.5), self.y_percent(616.5), self.scale_factor, anchor="center"),
+            "name": Textbox(self.screen, players[2][1], self.x_percent(1026.5), self.y_percent(697), 188, 72, font_size=40, scale_factor=self.scale_factor, text_color=WHITE, anchor="center"),
+            "status": Image(self.screen, self.pending_image, self.x_percent(1133.5), self.y_percent(503), self.scale_factor, anchor="center")
+        }
+
+    def create_four_profile(self, players):
+        """Create UI elements for a Four player profile.
+
+        Args:
+            players (list of tuples): A list containing one player tuple in the format (player_id, player_name).
+        """
+        self.profiles[players[0][0]] = {
+            "profile": Image(self.screen, self.profile_image, self.x_percent(239.5), self.y_percent(616.5), self.scale_factor, anchor="center"),
+            "name": Textbox(self.screen, players[0][1], self.x_percent(239.5), self.y_percent(697), 188, 72, font_size=40, scale_factor=self.scale_factor, text_color=WHITE, anchor="center"),
+            "status": Image(self.screen, self.pending_image, self.x_percent(347), self.y_percent(503), self.scale_factor, anchor="center")
+        }
+        self.profiles[players[1][0]] = {
+            "profile": Image(self.screen, self.profile_image, self.x_percent(552.5), self.y_percent(616.5), self.scale_factor, anchor="center"),
+            "name": Textbox(self.screen, players[1][1], self.x_percent(552.5), self.y_percent(697), 188, 72, font_size=40, scale_factor=self.scale_factor, text_color=WHITE, anchor="center"),
+            "status": Image(self.screen, self.pending_image, self.x_percent(660), self.y_percent(503), self.scale_factor, anchor="center")
+        }
+        self.profiles[players[2][0]] = {
+            "profile": Image(self.screen, self.profile_image, self.x_percent(865.5), self.y_percent(616.5), self.scale_factor, anchor="center"),
+            "name": Textbox(self.screen,players[2][1], self.x_percent(865.5), self.y_percent(697), 188, 72, font_size=40, scale_factor=self.scale_factor, text_color=WHITE, anchor="center"),
+            "status": Image(self.screen, self.pending_image, self.x_percent(973), self.y_percent(503), self.scale_factor, anchor="center")
+        }
+        self.profiles[players[3][0]] = {
+            "profile": Image(self.screen, self.profile_image, self.x_percent(1178.5), self.y_percent(616.5), self.scale_factor, anchor="center"),
+            "name": Textbox(self.screen, players[3][1], self.x_percent(1178.5), self.y_percent(697), 188, 72, font_size=40, scale_factor=self.scale_factor, text_color=WHITE, anchor="center"),
+            "status": Image(self.screen, self.pending_image, self.x_percent(1286), self.y_percent(503), self.scale_factor, anchor="center")
+        }
+
+    def create_profile(self, players):
```

Comment:
Let's predefine the `x, y, spacing_offset` for these 4 cases to cut down on the file size.